app/extensions/logging/__init__.py

```diff
@@ -4,45 +4,58 @@
 ---------------
 """
 import logging
-import enum
 from flask_login import current_user  # NOQA
+import enum
 
 
 class AuditType(str, enum.Enum):
     Create = 'Create'
     Delete = 'Delete'
     Update = 'Update'  # Generic Update
+    Fault = 'Fault'
+    Other = 'Other'  # None of the above
 
 
-# somewhere between Error and Critical
+# somewhere between Error and Critical to guarantee that it appears in the logs but is not interpreted as a
+# real error by the reader
 AUDIT = 45
 
 
 # logger for calling file needed as a parameter to ensure that the file and line numbers are correct in logs
-def audit_log(logger, msg, *args, **kwargs):
+def audit_log(logger, msg, audit_type=AuditType.Other, *args, **kwargs):
     assert object
 
-    # First iteration. Timestamp added by logger so no need to add manually
+    user_email = 'anonymous user'
     if current_user and not current_user.is_anonymous:
         msg = f'{msg} executed by user :{current_user.guid} {current_user.email}'
+        user_email = current_user.email
     else:
         msg = f' {msg} executed by anonymous user'
     logger.log(AUDIT, msg, *args, **kwargs)
+    from app.modules.audit_logs.models import AuditLog
```

Comment:
Anyone any idea why this is triggering a scary code scanning warning.  
It seems to be warning about uuids being logged but that's exactly what we need.

Reply:
Yeah, I'll dismiss those. It's working as intended.